shared/middleware/table_coerce.rb

```diff
@@ -0,0 +1,105 @@
+# A set of methods for coercing data in our table to specific types. In general,
+# the goal when converting is to change values where we can, but leave alone
+# those where we don't know how to convert (i.e. I can't convert "123" to a
+# boolean.
+
+module TableCoerce
+  # Given a set of records, and a list of columns, attempt to convert each
+  # column to a particular type. If we can convert the entire column to
+  # number or boolean, we'll do that, otherwise we'll convert all values to strings.
+  def TableCoerce.coerce(records, columns)
```

Comment:
I am struggling to think of a name that will make it a little more obvious what this method does. what do you think about calling this `coerce_columns_from_data` ?